control-plane/roles/monitoring/tasks/main.yaml

```diff
@@ -0,0 +1,106 @@
+        labels:
+          name: "{{ metal_monitoring_namespace }}"
+    
+- name: Deploy kube-prometheus-stack
+  include_role:
+    name: ansible-common/roles/helm-chart
+  vars:
+    helm_repo: "https://prometheus-community.github.io/helm-charts"
+    helm_chart: kube-prometheus-stack
+    helm_target_namespace: "{{ metal_monitoring_namespace }}"
+    helm_release_name: kube-prometheus-stack
+    helm_chart_version: "{{ prometheus_chart_version }}"
+    helm_value_file_template: "prometheus-stack-values.yaml"
+
+- name: Deploy Grafana Dashboard ConfigMaps
+  k8s:
+    definition: "{{ lookup('template', 'grafana_dashboards/' + item) }}"
+    namespace: "{{ metal_monitoring_namespace }}"
+  loop:
+     - go-processes.yaml
+     - alertmanager.yaml
+     - ipmi-state.yaml
+     - frr-state.yaml
+     - cumulus-state.yaml
+     - metal-api-state.yaml
+     - network-traffic.yaml
+     - nginx.yaml
+     - rgw-usage.yaml
+     - fio-exporter.yaml
+     - machine-capacity.yaml
+     - leaf-switches.yaml
+
+- name: Deploy Grafana dashboards
+  k8s:
+    definition: "{{ lookup('template', 'grafana_dashboards/' + item) }}"
+    namespace: "{{ metal_monitoring_namespace }}"
+  loop:
+    - go-processes.yaml
+    - metal-api.yaml
+
+
+- name: Deploy metal-metrics-exporter
+  k8s:
+    definition: "{{ lookup('template', 'metal-metrics-exporter/' + item) }}"
+    namespace: "{{ metal_monitoring_namespace }}"
+  loop:
+    - metal-metrics-exporter-secret.yaml
+    - metal-metrics-exporter-deployment.yaml
+    - metal-metrics-exporter-service.yaml
+    - metal-metrics-exporter-servicemonitor.yaml
+
+
+- name: Create service monitors
+  k8s:
+    definition: "{{ lookup('template', 'servicemonitors/' + item) }}"
+    namespace: "{{ metal_monitoring_namespace }}"
+  loop:
+    - metal-metrics-exporter-servicemonitor.yaml
+    - machine_controller_manager_servicemonitor.yaml
+    - metal_api_servicemonitor.yaml
+    - metal_db_servicemonitor.yaml
+    - masterdata_api_servicemonitor.yaml
+    - masterdata_db_servicemonitor.yaml
+    - ipam_db_servicemonitor.yaml
+    - nginx_servicemonitor.yaml
+
+
+- name: Create monitoring ingress for grafana
+  k8s:
+    definition: "{{ lookup('template', 'grafana_ingress.yaml') }}"
+    namespace: "{{ metal_monitoring_namespace }}"
+    apply: true
+
+- name: Deploy RethinkDB Exporter
+  k8s:
+    definition: "{{ lookup('template', item) }}"
+    namespace: "{{ metal_monitoring_namespace }}"
+  loop:
+     - rethinkdb-exporter.yaml
+
+
+- name: Deploy gardener-metrics-exporter
+  k8s:
+    definition: "{{ lookup('template', 'gardener-metrics-exporter/' + item) }}"
+    namespace: "{{ metal_monitoring_namespace }}"
+  loop:
+     - secret-metric-exporter.yaml
+     - serviceaccount-metrics-exporter.yaml
+     - rbac.yaml
+     - deployment-metrics-exporter.yaml
+     - service-metrics-exporter.yaml
+  when: lookup('k8s', api_version='v1', kind='Namespace', resource_name='garden')
```

Comment:
Skip gardener for now.

Reply:
https://github.com/metal-stack/metal-roles/commit/7d3f874375121e97e83afb8f079104299fd59183